Lib/hmac.py

```diff
@@ -179,9 +193,11 @@ def digest(key, msg, digest):
             A hashlib constructor returning a new hash object. *OR*
             A module supporting PEP 247.
     """
-    if (_hashopenssl is not None and
-            isinstance(digest, str) and digest in _openssl_md_meths):
-        return _hashopenssl.hmac_digest(key, msg, digest)
+    if _hashopenssl is not None and isinstance(digest, (str, _functype)):
+        try:
+            return _hashopenssl.hmac_digest(key, msg, digest)
+        except ValueError:
+            pass
```

Comment:
existing behavior since `hmac.digest()` was introduced in 3.7... but it is very odd that we do not either fall back to the slow path or raise an exception when `_hashopenssl` does not work.

The docstring and the hmac module docs never mention that this function could fail and will simply return None instead of bytes.

I'd rather it never return None.  just make the last line of the body be the traditional slower path:
`return HMAC(key, msg, digest).digest()`

That's what the user wants anyways.  (arguably fixing this could be its own PR - your choice)

Reply:
I'm confused. Did you review an intermediate commit perhaps? I don't see any code path that can return ``None``. ``hmac_digest()`` either returns a value or fails with an exception.

* if digest is a valid hash name or a ``_hashlib.openssl_foo`` function object, the function uses ``hmac_digest()``.
* if digest is another callable or module with ``new``, the fallback code path is used.
* if function is an invalid hash name, ``hmac_digest()`` or ``_hashlib.new()`` raises a ValueError

Reply:
github UX vs PEBKAC... somehow I didn't see the code below this in the function. :)